### PR TITLE
[82293] Implement loading state for huggingface grid

### PR DESCRIPTION
**Pull request by as-suvorov**
### All Submissions:
* [x] Create a meaningful name of the PR with reference to JIRA. Example: *[43714] [NIGHTLY] Take correct tag for snyk scans*
* [x] JIRA tickets:
  * 82293 - Implement loading state for huggingface grid
* [x] Set assignee to the PR.
* [x] Add corresponding labels to the PR. 
* [x] Set appropriate milestone to the PR.
* [ ] Added tests for new functionality (unit / e2e)
* [x] Followed code style guidelines:
  * Code covered with typings
  * No debug printing in production code
  * No commented blocks of code in production code

### Available commands in pull request comments

1. `/bump:client` - bump to latest **master** branch in **client** submodule
1. `/bump:client:releases/2021/4` - bump to latest **releases/2021/4** branch in **client** submodule
1. `/bump:client:yatarkan:feature/yt/feature-name` - bump to latest **feature/yt/feature-name** branch from **yatarkan** remote fork in **client** submodule
1. `/update` - merge base (target) branch into target branch and push changes

> **NOTE**:
> For more details, please visit the [DEVELOPER.md](development/docs/DEVELOPER.md)

### CODE REVIEW

File: client/src/app/root-store/huggingface-model-store/huggingface-model-store.state.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+import { ErrorState } from '@store/state';
+
+export interface State {
+  modelsData: IHFModelsData;
+  isLoading: boolean;
+  error: ErrorState;
+  modelReadme: string;
+  isLoadingReadme: boolean;
+  modelReadmeError: ErrorState;
```
Comment (yatarkan) [inline client/src/app/root-store/huggingface-model-store/huggingface-model-store.state.ts:11]: How about differentiating loading and error fields, e.g.
1. By naming
```suggestion
  modelsData: IHFModelsData;
  isModelsDataLoading: boolean;
  modelsDataError: ErrorState;
  modelReadme: string;
  isLoadingReadme: boolean;
  modelReadmeError: ErrorState;
```
2. Or by adding one more level for readme
```suggestion
  modelsData: IHFModelsData;
  isLoading: boolean;
  error: ErrorState;
  selectedModelDetails: {
    readme: string;
    isLoading: boolean;
    error: ErrorState;
  },
  modelReadme: string;
  isLoadingReadme: boolean;
  modelReadmeError: ErrorState;
```

**Review by yatarkan — COMMENTED** [review]
LGTM in general, a couple of comments

File: client/src/app/root-store/huggingface-model-store/huggingface-model-store.actions.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { createAction, props } from '@ngrx/store';
+
+import { IHFModelsData } from '@core/services/api/rest/huggingface.service';
+
+import { ErrorState } from '@store/state';
+
+export const loadModelData = createAction('[Huggingface model] Load model data');
+export const loadModelDataSuccess = createAction(
+  '[Huggingface model] Load model data success',
```
[the remark this answers is not in the document]
Comment (as-suvorov) [reply client/src/app/root-store/huggingface-model-store/huggingface-model-store.actions.ts:9]: discussed

File: client/src/app/root-store/huggingface-model-store/huggingface-model-store.state.ts (hunk 1 above)
Comment (as-suvorov) [reply client/src/app/root-store/huggingface-model-store/huggingface-model-store.state.ts:11]: fixed

**Review by vishniakov-nikolai — APPROVED** [review]

**Review by yatarkan — APPROVED** [review]

**Pull request merged**